dtworkload's perf and random test run failure

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
run command `bash-5.1# /tiflash/tiflash dtworkload --testing_type=daily_perf --verify_round=100 --log_file=/tmp/dt.log`

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
command run success

### 3. What did you see instead (Required)
```
bash-5.1# /tiflash/tiflash dtworkload --testing_type=daily_perf --verify_round=100 --log_file=/tmp/dt.log
Date,Table Schema,Workload,Init Seconds,Write Speed(rows count),Read Speed(rows count)

libc++abi: terminating with uncaught exception of type DB::Exception: DB::Exception
Aborted
```
```
[92m[2023/05/25 06:00:49.450 +00:00] [0m[1m[31m[ERROR] [0m[35m[WriteBatchesImpl.h:72] [0m[31m["!!!=========================Modifications in log haven't persisted=========================!!! Stack trace: \n       0x735bf0c\tDB::DM::WriteBatches::~WriteBatches()::'lambda'(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::operator()(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const [tiflash+120962828]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:68\n       0x734a6d3\tDB::DM::WriteBatches::~WriteBatches() [tiflash+120891091]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:75\n       0x74f61b7\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122642871]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:376\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x736f4d3\tDB::DM::DeltaMergeStore::handleBackgroundTask(bool) [tiflash+121042131]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:390\n       0x7e4160b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, DB::BackgroundProcessingPool::BackgroundProcessingPool(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)::$_1> >(void*) [tiflash+132388363]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[thread_id=31][0m
[92m[2023/05/25 06:00:49.450 +00:00] [0m[1m[31m[ERROR] [0m[35m[WriteBatchesImpl.h:72] [0m[31m["!!!=========================Modifications in log haven't persisted=========================!!! Stack trace: \n       0x735bf0c\tDB::DM::WriteBatches::~WriteBatches()::'lambda'(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::operator()(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const [tiflash+120962828]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:68\n       0x734a6d3\tDB::DM::WriteBatches::~WriteBatches() [tiflash+120891091]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:75\n       0x74f61b7\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122642871]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:376\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x736f4d3\tDB::DM::DeltaMergeStore::handleBackgroundTask(bool) [tiflash+121042131]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:390\n       0x7e4160b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, DB::BackgroundProcessingPool::BackgroundProcessingPool(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)::$_1> >(void*) [tiflash+132388363]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[thread_id=32][0m
[92m[2023/05/25 06:00:49.450 +00:00] [0m[1m[31m[ERROR] [0m[35m[WriteBatchesImpl.h:72] [0m[31m["!!!=========================Modifications in log haven't persisted=========================!!! Stack trace: \n       0x735bf0c\tDB::DM::WriteBatches::~WriteBatches()::'lambda'(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::operator()(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const [tiflash+120962828]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:68\n       0x734a6d3\tDB::DM::WriteBatches::~WriteBatches() [tiflash+120891091]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:75\n       0x74f61b7\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122642871]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:376\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x7351802\tDB::DM::DeltaMergeStore::checkSegmentUpdate(std::__1::shared_ptr<DB::DM::DMContext> const&, std::__1::shared_ptr<DB::DM::Segment> const&, DB::DM::DeltaMergeStore::ThreadType) [tiflash+120920066]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1424\n       0x736f688\tDB::DM::DeltaMergeStore::handleBackgroundTask(bool) [tiflash+121042568]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:421\n       0x7e4160b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, DB::BackgroundProcessingPool::BackgroundProcessingPool(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)::$_1> >(void*) [tiflash+132388363]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[thread_id=39][0m
[92m[2023/05/25 06:00:49.450 +00:00] [0m[1m[31m[ERROR] [0m[35m[DeltaMergeStore_InternalBg.cpp:410] [0m[31m["Execute task on segment failed, task=Flush segment=<segment_id=1 epoch=14 range=[-9223372036854775808,5004489) abandoned=true> err=try to create entry version with invalid state [ver=11574.0] [entry=PageEntry{file: 1, offset: 0xC61F811, size: 160790, checksum: 0xD3CC0A8F947E4C13, tag: 0, field_offsets: [0,51,2957,8305,17727,27320,30232,35580,45002,54595,64015,73605,86818,96277,105860,111205,113925,124104,144258], checkpoint_info: invalid}] [state={type:VAR_REF, create_ver: 141.0, is_deleted: true, delete_ver: 341.0, ori_page_id: 0.72, being_ref_count: 1, num_entries: 0}]:  [type=PUT] [page_id=0.105] [ver=11574.0] [edit_size=3]"] [0m[90m[source="keyspace_id=4294967295 table_id=0"] [thread_id=31][0m
[92m[2023/05/25 06:00:49.450 +00:00] [0m[1m[31m[ERROR] [0m[35m[DeltaMergeStore_InternalBg.cpp:410] [0m[31m["Execute task on segment failed, task=Flush segment=<segment_id=11 epoch=2 range=[14996219,9223372036854775807) abandoned=true> err=try to create entry version with invalid state [ver=11573.0] [entry=PageEntry{file: 1, offset: 0xC65CC0B, size: 160546, checksum: 0x655AB76DAB7D6F14, tag: 0, field_offsets: [0,51,2981,8301,17686,27243,30189,35509,44894,54451,63835,73390,86553,96116,105664,110978,113680,123852,144023], checkpoint_info: invalid}] [state={type:VAR_REF, create_ver: 143.0, is_deleted: true, delete_ver: 343.0, ori_page_id: 0.74, being_ref_count: 1, num_entries: 0}]:  [type=PUT] [page_id=0.107] [ver=11573.0] [edit_size=2]"] [0m[90m[source="keyspace_id=4294967295 table_id=0"] [thread_id=32][0m
[92m[2023/05/25 06:00:49.458 +00:00] [0m[1m[94m[INFO] [0m[35m[DTWorkload.cpp:275] [0m["scanAll: columns 19 streams 4 read_stat ms 327 count 4766655 speed 14576926"] [0m[90m[source=DTWorkload] [thread_id=38][0m
[92m[2023/05/25 06:00:49.467 +00:00] [0m[1m[31m[ERROR] [0m[35m[WriteBatchesImpl.h:72] [0m[31m["!!!=========================Modifications in log haven't persisted=========================!!! Stack trace: \n       0x735bf0c\tDB::DM::WriteBatches::~WriteBatches()::'lambda'(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::operator()(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const [tiflash+120962828]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:68\n       0x734a6d3\tDB::DM::WriteBatches::~WriteBatches() [tiflash+120891091]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:75\n       0x74f61b7\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122642871]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:376\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x7351802\tDB::DM::DeltaMergeStore::checkSegmentUpdate(std::__1::shared_ptr<DB::DM::DMContext> const&, std::__1::shared_ptr<DB::DM::Segment> const&, DB::DM::DeltaMergeStore::ThreadType) [tiflash+120920066]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1424\n       0x734dd14\tDB::DM::DeltaMergeStore::write(DB::Context const&, DB::Settings const&, DB::Block&) [tiflash+120904980]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:675\n       0x1daf9a3\tDB::DM::tests::DTWorkload::write(DB::DM::tests::ThreadStat&) [tiflash+31127971]\n                \tdbms/src/Storages/DeltaMerge/workload/DTWorkload.cpp:149\n       0x1db4666\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::DM::tests::DTWorkload::*)(DB::DM::tests::ThreadStat&), DB::DM::tests::DTWorkload*, std::__1::reference_wrapper<DB::DM::tests::ThreadStat> > >(void*) [tiflash+31147622]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[thread_id=40][0m
[92m[2023/05/25 06:00:49.468 +00:00] [0m[1m[31m[ERROR] [0m[35m[Exception.cpp:90] [0m[31m["Code: 10017, e.displayText() = DB::Exception: try to create entry version with invalid state [ver=11574.0] [entry=PageEntry{file: 1, offset: 0xC61F811, size: 160790, checksum: 0xD3CC0A8F947E4C13, tag: 0, field_offsets: [0,51,2957,8305,17727,27320,30232,35580,45002,54595,64015,73605,86818,96277,105860,111205,113925,124104,144258], checkpoint_info: invalid}] [state={type:VAR_REF, create_ver: 141.0, is_deleted: true, delete_ver: 341.0, ori_page_id: 0.72, being_ref_count: 1, num_entries: 0}]:  [type=PUT] [page_id=0.105] [ver=11574.0] [edit_size=3], e.what() = DB::Exception, Stack trace:\n\n\n       0x1c197ee\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [tiflash+29464558]\n                \tdbms/src/Common/Exception.h:46\n       0x1b3b947\tDB::PS::V3::VersionedPageEntries<DB::PS::V3::u128::PageDirectoryTrait>::createNewEntry(DB::PS::V3::PageVersion const&, DB::PS::V3::PageEntryV3 const&) [tiflash+28555591]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:120\n       0x1b49da0\tDB::PS::V3::PageDirectory<DB::PS::V3::u128::PageDirectoryTrait>::apply(DB::PS::V3::PageEntriesEdit<DB::UInt128>&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+28614048]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:1588\n       0x75c6b74\tDB::PS::V3::PageStorageImpl::writeImpl(DB::WriteBatch&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+123497332]\n                \tdbms/src/Storages/Page/V3/PageStorageImpl.cpp:139\n       0x75743d7\tDB::PageWriter::write(DB::WriteBatchWrapper&&, std::__1::shared_ptr<DB::WriteLimiter>) const [tiflash+123159511]\n                \tdbms/src/Storages/Page/PageStorage.cpp:580\n       0x734fec8\tDB::DM::WriteBatches::writeLogAndData() [tiflash+120913608]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:143\n       0x7503119\tDB::DM::ColumnFileFlushTask::prepare(DB::DM::WriteBatches&) [tiflash+122695961]\n                \tdbms/src/Storages/DeltaMerge/Delta/ColumnFileFlushTask.cpp:52\n       0x74f4b47\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122637127]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:334\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x736f4d3\tDB::DM::DeltaMergeStore::handleBackgroundTask(bool) [tiflash+121042131]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:390\n       0x7e4160b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, DB::BackgroundProcessingPool::BackgroundProcessingPool(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)::$_1> >(void*) [tiflash+132388363]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[source="void DB::BackgroundProcessingPool::threadFunction(size_t)"] [thread_id=31][0m
[92m[2023/05/25 06:00:49.468 +00:00] [0m[1m[31m[ERROR] [0m[35m[Exception.cpp:90] [0m[31m["Code: 10017, e.displayText() = DB::Exception: try to create entry version with invalid state [ver=11573.0] [entry=PageEntry{file: 1, offset: 0xC65CC0B, size: 160546, checksum: 0x655AB76DAB7D6F14, tag: 0, field_offsets: [0,51,2981,8301,17686,27243,30189,35509,44894,54451,63835,73390,86553,96116,105664,110978,113680,123852,144023], checkpoint_info: invalid}] [state={type:VAR_REF, create_ver: 143.0, is_deleted: true, delete_ver: 343.0, ori_page_id: 0.74, being_ref_count: 1, num_entries: 0}]:  [type=PUT] [page_id=0.107] [ver=11573.0] [edit_size=2], e.what() = DB::Exception, Stack trace:\n\n\n       0x1c197ee\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [tiflash+29464558]\n                \tdbms/src/Common/Exception.h:46\n       0x1b3b947\tDB::PS::V3::VersionedPageEntries<DB::PS::V3::u128::PageDirectoryTrait>::createNewEntry(DB::PS::V3::PageVersion const&, DB::PS::V3::PageEntryV3 const&) [tiflash+28555591]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:120\n       0x1b49da0\tDB::PS::V3::PageDirectory<DB::PS::V3::u128::PageDirectoryTrait>::apply(DB::PS::V3::PageEntriesEdit<DB::UInt128>&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+28614048]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:1588\n       0x75c6b74\tDB::PS::V3::PageStorageImpl::writeImpl(DB::WriteBatch&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+123497332]\n                \tdbms/src/Storages/Page/V3/PageStorageImpl.cpp:139\n       0x75743d7\tDB::PageWriter::write(DB::WriteBatchWrapper&&, std::__1::shared_ptr<DB::WriteLimiter>) const [tiflash+123159511]\n                \tdbms/src/Storages/Page/PageStorage.cpp:580\n       0x734fec8\tDB::DM::WriteBatches::writeLogAndData() [tiflash+120913608]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:143\n       0x7503119\tDB::DM::ColumnFileFlushTask::prepare(DB::DM::WriteBatches&) [tiflash+122695961]\n                \tdbms/src/Storages/DeltaMerge/Delta/ColumnFileFlushTask.cpp:52\n       0x74f4b47\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122637127]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:334\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x736f4d3\tDB::DM::DeltaMergeStore::handleBackgroundTask(bool) [tiflash+121042131]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:390\n       0x7e4160b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, DB::BackgroundProcessingPool::BackgroundProcessingPool(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)::$_1> >(void*) [tiflash+132388363]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[source="void DB::BackgroundProcessingPool::threadFunction(size_t)"] [thread_id=32][0m
[92m[2023/05/25 06:00:49.468 +00:00] [0m[1m[31m[ERROR] [0m[35m[WriteBatchesImpl.h:72] [0m[31m["!!!=========================Modifications in log haven't persisted=========================!!! Stack trace: \n       0x735bf0c\tDB::DM::WriteBatches::~WriteBatches()::'lambda'(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)::operator()(DB::WriteBatchWrapper const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const [tiflash+120962828]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:68\n       0x734a6d3\tDB::DM::WriteBatches::~WriteBatches() [tiflash+120891091]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:75\n       0x74f61b7\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122642871]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:376\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x736f4d3\tDB::DM::DeltaMergeStore::handleBackgroundTask(bool) [tiflash+121042131]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:390\n       0x7e4160b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, DB::BackgroundProcessingPool::BackgroundProcessingPool(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)::$_1> >(void*) [tiflash+132388363]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[thread_id=30][0m
[92m[2023/05/25 06:00:49.468 +00:00] [0m[1m[31m[ERROR] [0m[35m[Exception.cpp:90] [0m[31m["Code: 10017, e.displayText() = DB::Exception: try to create entry version with invalid state [ver=11573.0] [entry=PageEntry{file: 1, offset: 0xC6A950A, size: 27431, checksum: 0x5206A2FE33AAA4B0, tag: 0, field_offsets: [0,38,583,1471,3039,4654,5195,6083,7651,9266,10833,12444,15039,16642,18254,19138,19618,21327,24739], checkpoint_info: invalid}] [state={type:VAR_REF, create_ver: 145.0, is_deleted: true, delete_ver: 345.0, ori_page_id: 0.76, being_ref_count: 1, num_entries: 0}]:  [type=PUT] [page_id=0.109] [ver=11573.0] [edit_size=5], e.what() = DB::Exception, Stack trace:\n\n\n       0x1c197ee\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [tiflash+29464558]\n                \tdbms/src/Common/Exception.h:46\n       0x1b3b947\tDB::PS::V3::VersionedPageEntries<DB::PS::V3::u128::PageDirectoryTrait>::createNewEntry(DB::PS::V3::PageVersion const&, DB::PS::V3::PageEntryV3 const&) [tiflash+28555591]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:120\n       0x1b49da0\tDB::PS::V3::PageDirectory<DB::PS::V3::u128::PageDirectoryTrait>::apply(DB::PS::V3::PageEntriesEdit<DB::UInt128>&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+28614048]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:1588\n       0x75c6b74\tDB::PS::V3::PageStorageImpl::writeImpl(DB::WriteBatch&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+123497332]\n                \tdbms/src/Storages/Page/V3/PageStorageImpl.cpp:139\n       0x75743d7\tDB::PageWriter::write(DB::WriteBatchWrapper&&, std::__1::shared_ptr<DB::WriteLimiter>) const [tiflash+123159511]\n                \tdbms/src/Storages/Page/PageStorage.cpp:580\n       0x734fec8\tDB::DM::WriteBatches::writeLogAndData() [tiflash+120913608]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:143\n       0x7503119\tDB::DM::ColumnFileFlushTask::prepare(DB::DM::WriteBatches&) [tiflash+122695961]\n                \tdbms/src/Storages/DeltaMerge/Delta/ColumnFileFlushTask.cpp:52\n       0x74f4b47\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122637127]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:334\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x7351802\tDB::DM::DeltaMergeStore::checkSegmentUpdate(std::__1::shared_ptr<DB::DM::DMContext> const&, std::__1::shared_ptr<DB::DM::Segment> const&, DB::DM::DeltaMergeStore::ThreadType) [tiflash+120920066]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1424\n       0x736f688\tDB::DM::DeltaMergeStore::handleBackgroundTask(bool) [tiflash+121042568]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:421\n       0x7e4160b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, DB::BackgroundProcessingPool::BackgroundProcessingPool(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)::$_1> >(void*) [tiflash+132388363]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[source="void DB::BackgroundProcessingPool::threadFunction(size_t)"] [thread_id=39][0m
[92m[2023/05/25 06:00:49.468 +00:00] [0m[1m[31m[ERROR] [0m[35m[DeltaMergeStore_InternalBg.cpp:410] [0m[31m["Execute task on segment failed, task=Flush segment=<segment_id=9 epoch=3 range=[5004489,10003426)> err=try to create entry version with invalid state [ver=11595.0] [entry=PageEntry{file: 1, offset: 0xC8C0579, size: 159417, checksum: 0x383EDC9A19598BC5, tag: 0, field_offsets: [0,51,2946,8290,17549,26998,29898,35242,44501,53950,63204,72648,85661,95137,104576,109914,112647,122705,142886], checkpoint_info: invalid}] [state={type:VAR_REF, create_ver: 161.0, is_deleted: true, delete_ver: 361.0, ori_page_id: 0.92, being_ref_count: 1, num_entries: 0}]:  [type=PUT] [page_id=0.125] [ver=11595.0] [edit_size=3]"] [0m[90m[source="keyspace_id=4294967295 table_id=0"] [thread_id=30][0m
[92m[2023/05/25 06:00:49.468 +00:00] [0m[1m[31m[ERROR] [0m[35m[Exception.cpp:90] [0m[31m["Code: 10017, e.displayText() = DB::Exception: try to create entry version with invalid state [ver=11595.0] [entry=PageEntry{file: 1, offset: 0xC8C0579, size: 159417, checksum: 0x383EDC9A19598BC5, tag: 0, field_offsets: [0,51,2946,8290,17549,26998,29898,35242,44501,53950,63204,72648,85661,95137,104576,109914,112647,122705,142886], checkpoint_info: invalid}] [state={type:VAR_REF, create_ver: 161.0, is_deleted: true, delete_ver: 361.0, ori_page_id: 0.92, being_ref_count: 1, num_entries: 0}]:  [type=PUT] [page_id=0.125] [ver=11595.0] [edit_size=3], e.what() = DB::Exception, Stack trace:\n\n\n       0x1c197ee\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [tiflash+29464558]\n                \tdbms/src/Common/Exception.h:46\n       0x1b3b947\tDB::PS::V3::VersionedPageEntries<DB::PS::V3::u128::PageDirectoryTrait>::createNewEntry(DB::PS::V3::PageVersion const&, DB::PS::V3::PageEntryV3 const&) [tiflash+28555591]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:120\n       0x1b49da0\tDB::PS::V3::PageDirectory<DB::PS::V3::u128::PageDirectoryTrait>::apply(DB::PS::V3::PageEntriesEdit<DB::UInt128>&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+28614048]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:1588\n       0x75c6b74\tDB::PS::V3::PageStorageImpl::writeImpl(DB::WriteBatch&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+123497332]\n                \tdbms/src/Storages/Page/V3/PageStorageImpl.cpp:139\n       0x75743d7\tDB::PageWriter::write(DB::WriteBatchWrapper&&, std::__1::shared_ptr<DB::WriteLimiter>) const [tiflash+123159511]\n                \tdbms/src/Storages/Page/PageStorage.cpp:580\n       0x734fec8\tDB::DM::WriteBatches::writeLogAndData() [tiflash+120913608]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:143\n       0x7503119\tDB::DM::ColumnFileFlushTask::prepare(DB::DM::WriteBatches&) [tiflash+122695961]\n                \tdbms/src/Storages/DeltaMerge/Delta/ColumnFileFlushTask.cpp:52\n       0x74f4b47\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122637127]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:334\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x736f4d3\tDB::DM::DeltaMergeStore::handleBackgroundTask(bool) [tiflash+121042131]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore_InternalBg.cpp:390\n       0x7e4160b\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, DB::BackgroundProcessingPool::BackgroundProcessingPool(int, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >)::$_1> >(void*) [tiflash+132388363]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[source="void DB::BackgroundProcessingPool::threadFunction(size_t)"] [thread_id=30][0m
[92m[2023/05/25 06:00:49.468 +00:00] [0m[1m[31m[ERROR] [0m[35m[Exception.cpp:90] [0m[31m["Code: 10017, e.displayText() = DB::Exception: try to create entry version with invalid state [ver=11573.0] [entry=PageEntry{file: 1, offset: 0xC73AF2A, size: 9667, checksum: 0x26D8C5CEB1DF7438, tag: 0, field_offsets: [0,36,245,560,1064,1602,1811,2126,2630,3168,3671,4207,5385,5918,6446,6758,6939,7523,8738], checkpoint_info: invalid}] [state={type:VAR_REF, create_ver: 150.0, is_deleted: true, delete_ver: 350.0, ori_page_id: 0.81, being_ref_count: 1, num_entries: 0}]:  [type=PUT] [page_id=0.114] [ver=11573.0] [edit_size=11], e.what() = DB::Exception, Stack trace:\n\n\n       0x1c197ee\tDB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) [tiflash+29464558]\n                \tdbms/src/Common/Exception.h:46\n       0x1b3b947\tDB::PS::V3::VersionedPageEntries<DB::PS::V3::u128::PageDirectoryTrait>::createNewEntry(DB::PS::V3::PageVersion const&, DB::PS::V3::PageEntryV3 const&) [tiflash+28555591]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:120\n       0x1b49da0\tDB::PS::V3::PageDirectory<DB::PS::V3::u128::PageDirectoryTrait>::apply(DB::PS::V3::PageEntriesEdit<DB::UInt128>&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+28614048]\n                \tdbms/src/Storages/Page/V3/PageDirectory.cpp:1588\n       0x75c6b74\tDB::PS::V3::PageStorageImpl::writeImpl(DB::WriteBatch&&, std::__1::shared_ptr<DB::WriteLimiter> const&) [tiflash+123497332]\n                \tdbms/src/Storages/Page/V3/PageStorageImpl.cpp:139\n       0x75743d7\tDB::PageWriter::write(DB::WriteBatchWrapper&&, std::__1::shared_ptr<DB::WriteLimiter>) const [tiflash+123159511]\n                \tdbms/src/Storages/Page/PageStorage.cpp:580\n       0x734fec8\tDB::DM::WriteBatches::writeLogAndData() [tiflash+120913608]\n                \tdbms/src/Storages/DeltaMerge/WriteBatchesImpl.h:143\n       0x7503119\tDB::DM::ColumnFileFlushTask::prepare(DB::DM::WriteBatches&) [tiflash+122695961]\n                \tdbms/src/Storages/DeltaMerge/Delta/ColumnFileFlushTask.cpp:52\n       0x74f4b47\tDB::DM::DeltaValueSpace::flush(DB::DM::DMContext&) [tiflash+122637127]\n                \tdbms/src/Storages/DeltaMerge/Delta/DeltaValueSpace.cpp:334\n       0x73bd5d3\tDB::DM::Segment::flushCache(DB::DM::DMContext&) [tiflash+121361875]\n                \tdbms/src/Storages/DeltaMerge/Segment.cpp:2016\n       0x7351802\tDB::DM::DeltaMergeStore::checkSegmentUpdate(std::__1::shared_ptr<DB::DM::DMContext> const&, std::__1::shared_ptr<DB::DM::Segment> const&, DB::DM::DeltaMergeStore::ThreadType) [tiflash+120920066]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:1424\n       0x734dd14\tDB::DM::DeltaMergeStore::write(DB::Context const&, DB::Settings const&, DB::Block&) [tiflash+120904980]\n                \tdbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp:675\n       0x1daf9a3\tDB::DM::tests::DTWorkload::write(DB::DM::tests::ThreadStat&) [tiflash+31127971]\n                \tdbms/src/Storages/DeltaMerge/workload/DTWorkload.cpp:149\n       0x1db4666\tvoid* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (DB::DM::tests::DTWorkload::*)(DB::DM::tests::ThreadStat&), DB::DM::tests::DTWorkload*, std::__1::reference_wrapper<DB::DM::tests::ThreadStat> > >(void*) [tiflash+31147622]\n                \t/usr/local/bin/../include/c++/v1/thread:291\n  0x7f561a38f802\tstart_thread [libc.so.6+653314]\n  0x7f561a32f314\t__GI___clone [libc.so.6+258836]"] [0m[90m[source="exception thrown in DTWorkload::write"] [thread_id=40][0m
```
### 4. What is your TiFlash version? (Required)
bash-5.1# /tiflash/tiflash version
TiFlash
Release Version: v7.2.0-alpha
Edition:         Community
Git Commit Hash: c1d2b83efe97cc106ba095dbd6ac8146de29552e
Git Branch:      heads/refs/tags/v7.2.0-alpha
UTC Build Time:  2023-05-19 11:34:10
Enable Features: jemalloc sm4(GmSSL) avx2 avx512 unwind thinlto
Profile:         RELWITHDEBINFO

Raft Proxy
Git Commit Hash:   5cdf47726c58e04683c2da3e2892c276fccb3e8a
Git Commit Branch: HEAD
UTC Build Time:    2023-05-19 11:40:33
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Storage Engine:    tiflash
Prometheus Prefix: tiflash_proxy_
Profile:           release
Enable Features:   Unknown (env var does not exist when building)

--- a/dbms/src/Storages/DeltaMerge/workload/DTWorkload.cpp
+++ b/dbms/src/Storages/DeltaMerge/workload/DTWorkload.cpp
@@ -51,6 +51,7 @@ DTWorkload::DTWorkload(const WorkloadOptions & opts_, std::shared_ptr<SharedHand
 
     key_gen = KeyGenerator::create(opts_);
     ts_gen = std::make_unique<TimestampGenerator>();
+    // max page id is only updated at restart, so we need recreate page v3 before recreate table
     context->initializeGlobalStoragePoolIfNeed(context->getPathPool());
     Stopwatch sw;
     store = std::make_unique<DeltaMergeStore>(

--- a/dbms/src/Storages/DeltaMerge/workload/DTWorkload.cpp
+++ b/dbms/src/Storages/DeltaMerge/workload/DTWorkload.cpp
@@ -51,7 +51,7 @@ DTWorkload::DTWorkload(const WorkloadOptions & opts_, std::shared_ptr<SharedHand
 
     key_gen = KeyGenerator::create(opts_);
     ts_gen = std::make_unique<TimestampGenerator>();
-
+    context->initializeGlobalStoragePoolIfNeed(context->getPathPool());
     Stopwatch sw;
     store = std::make_unique<DeltaMergeStore>(
         *context,

